package/kernel/mac80211/400-wifi-mac80211-do-not-offer-a-mesh-path-if-forwarding.patch

```diff
@@ -0,0 +1,61 @@
+From 0d47666f48a084363ee54f389bec2865de4934b8 Mon Sep 17 00:00:00 2001
+From: Benjamin Berg <benjamin@sipsolutions.net>
+Date: Wed, 30 Apr 2025 20:25:38 +0200
+Subject: [PATCH] wifi: mac80211: do not offer a mesh path if forwarding is
+ disabled
+
+When processing a PREQ the code would always check whether we have a
+mesh path locally and reply accordingly. However, when forwarding is
+disabled then we should not reply with this information as we will not
+forward data packets down that path.
+
+Move the check for dot11MeshForwarding up in the function and skip the
+mesh path lookup in that case. In the else block, set forward to false
+so that the rest of the function becomes a no-op and the
+dot11MeshForwarding check does not need to be duplicated.
+
+This explains an effect observed in the Freifunk community where mesh
+forwarding is disabled. In that case a mesh with three STAs and only bad
+links in between them, individual STAs would occionally have indirect
```

Comment:
> occionally

occasionally

Reply:
Updated it and moved the patch to the correct location. Still need to send a v2, but would prefer if we manage to test it more thoroughly before that.

Reply:
> individual STAs would occionally have indirect
mpath entries.

how can I look for these mpath entries in our Freifunk community in Aachen?
I would like to compare before and after

Reply:
Just do an "iw dev meshX mpath dump". I think that with the patch the PATH_CHANGE count should not keep increasing in the scenario. No update should be triggered as the intermediate STA/HOP will not send a response.

Also "iw phy phyX get txq" can show whether there are a lot of pending packets at the time.